### PR TITLE
Adding kafka producer error metrics to logger

### PR DESCRIPTION
**Pull request by kmurica**

### CODE REVIEW

**Comment by kmurica** [comment]
@matthew-d-jones Something like this? Corresponding Jira ticket:
https://jira.esss.lu.se/browse/ECDC-2317

File: forwarder/kafka/kafka_producer.py [hunk 1]
```diff
@@ -38,9 +39,13 @@ def ack(err, _):
                 topic, payload, key=key, on_delivery=ack, timestamp=timestamp_ms
             )
         except BufferError as e:
+            self._missed_data_count += 1
             self.logger.error(
                 "Producer message buffer is full. "
                 "Data loss occurred as messages are produced "
                 f"faster than are sent to the kafka broker: {e}"
             )
+            self.logger.error(
```
Comment (ebadkamil) [inline forwarder/kafka/kafka_producer.py:48]: May be combine this info in the previous log message. I am not sure.  What do you think?

**Review by matthew-d-jones — CHANGES_REQUESTED** [review]
Sorry I meant tracking with a counter that gets reported to graphite using what Ebad has implemented: https://github.com/ess-dmsc/forwarder/pull/82. It would allow us to track this error on a grafana dashboard.
It would also give us the option of not sending the log message altogether. My concern is that if we hit this error it could trigger at the rate we produce messages from the Forwarder (up to ~10,000 per second) and effectively mount a DoS attack on our logging server. The periodic reporting of the counter as a metric gets around this problem. We take a similar approach for errors which can be very frequent in the Filewriter.

**Review by matthew-d-jones — CHANGES_REQUESTED** [review]
Implementation looks good, but I think the severity is now not clear. `BufferError` specifically means the buffer is full, I wish it was called `FullBufferError`...
It is important that it is understood that this is a critical problem and that data is being lost. Could you use the original error log string as a comment where the `BufferError` is caught, and report the error to graphite as `data_loss_errors`?
```
                "Producer message buffer is full. "
                self._update_buffer_err_counter.increment()
                "Data loss occurred as messages are produced "
                f"faster than are sent to the kafka broker: {e}"
```

**Review by matthew-d-jones — APPROVED** [review]
Thanks, LGTM

**Comment by kmurica** [comment]
@ebadkamil does it look good to you? If so, could you merge the PR to main?

**Comment by ebadkamil** [comment]
yes Thanks. LGTM.

**Pull request merged**